terms/lstm_nm.tex

```diff
@@ -0,0 +1,124 @@
+
+In order to solve this problem, \citet{hochreiter1997long} proposed the Long Short-Term Memory network, or LSTM. 
+
+\section{LSTM Architecture}
+This module is a type of Recurrent Neural Network (RNN), that is, a repeated chained module, with a gradient-based unit applied. The main difference is that the repeated module itself is different, with the LSTM module being composed of a series of gates named the \textbf{input gate}, \textbf{input gate},   described in Figure~\ref{fig:lstm_architecture}.
+
+\begin{figure*}
+	\centering
+	\includegraphics[width=0.6\linewidth]{graphics/lstm/lstm.png}
+	\caption{
+		Architecture of standard LSTM module.
+	}
+	\label{fig:lstm_architecture}
+\end{figure*}
+
+The traversal from one gate to another is handled through a number of \textbf{activation functions}, which had been discussed in a previous chapter. The LSTM framework uses three main functions:
+
+{$\sigma _{g}$} - denoting the Sigmoid Function\index{Sigmoid Function} \citep{DavidE.Rumelhart1986Lrbb}.
+{$\sigma _{c}$} - denoting the Hyperbolic Tangent Function.\index{Hyperbolic Tangent Function}
+{$\sigma _{h}$} - also denotes the Hyperbolic Tangent Function in the case of the hidden gate, and implies that {$\sigma _{h}(x)=x$}.
+
+In order for the LSTM module to learn, backpropagation\index{LSTM Backpropagation} is implemented in the form of the Constant Error Carousel (CEC)\index{Constant Error Carousel} function, which updates the hidden gate {$h_{t}$} through the following equations:
+
+\begin{itemize}
+	\item {$h_{t}$} -  the hidden state, denoted by 
+	\begin{equation} \label{LSTM CEC}
+	h_{t} = o_{t} \circ tanh(c_{t})
+	\end{equation}
+	where {$\circ$} implies the Hadamard Product (element-wise multiplication).
+	\item {$o_{t}$} - the output gate, denoted by 
+	\begin{equation} \label{output gate equation}
+		\sigma(W_{xo}x_{t} + W_{ho}h_{t-1} + W_{co} \circ c_{t} + b_{o})
+	\end{equation}
+	\item {$c_{t}$} - the cell transfer state, denoted by 
+	\begin{equation} \label{cell state equation}
+		f_{t} \circ c_{t-1}+ i_{t} \circ tanh(W_{xc}x_{t} + W_{hc}h_{t-1} + b_{c})
+	\end{equation}
+	\item {$f_{t}$} - the forget gate, denoted by 
+	\begin{equation} \label{forget gate equation}
+		\sigma(W_{xf}x_{t} + W_{hf}h_{t-1} + W_{cf} \circ c_{t-1} + b_{f})
+	\end{equation}
+	\item {$i_{t}$} - the input gate, denoted by 
+	\begin{equation} \label{input gate equation}
+		\sigma(W_{xi}x_{t} + W_{hi}h_{t-1} + W_{ci} \circ c_{t-1} + b_{i})
+	\end{equation}
+	
+\end{itemize}
+
+These functions are not only the foundation of the LSTM module, but it is processed in such a way that it also reduced the Vanishing or Exploding Gradient Problem \index{Vanishing Gradient Problem} - the problem in which gradient descent begins to converge to zero or infinity, leaving the results almost unchanging in value.\citep{chapter-gradient-flow-2001}\\
```

Comment:
"reduceS" not reduced.  Remove manual breaks at end.  Citation before fullstop.  Please read guidelines in README